bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClientFactory.java

```diff
@@ -139,13 +144,33 @@ public boolean equals(Object o) {
     return false;
   }
 
-  private BigQueryReadClient createBigQueryReadClient(Optional<String> endpoint) {
+  private BigQueryReadClient createBigQueryReadClient(
+      Optional<String> endpoint, Optional<Integer> flowControlWindow) {
     try {
       InstantiatingGrpcChannelProvider.Builder transportBuilder = createTransportBuilder(endpoint);
+      if (flowControlWindow.isPresent()) {
+        ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder> channelConfigurator =
+            (ManagedChannelBuilder channelBuilder) -> {
```

Comment:
Yeah, I was surprised, there might have been something weird in my environment, I'll try this again with the simpler way.

Reply:
There are few instances of this class in the Intellij project as the provided and test dependencies are all together. In practice you can import `io.netty....` and it will work

Reply:
Tried this again and simpler code seems to work now.